torch/optim/rprop.py

```diff
@@ -307,7 +307,7 @@ def _view_complex_as_real(tensor_list):
                 grouped_prevs[i].copy_(grouped_grads[i])
         grouped_grads = grouped_prevs
 
-        signs = [s.sign() for s in signs]
+        signs = torch._foreach_sign(signs)
```

Comment:
beautiful, though could we do
```suggestion
        torch._foreach_sign_(signs)
```